src/vaadin-combo-box-mixin.html

```diff
@@ -352,8 +352,9 @@
     _onClick(e) {
       this._closeOnBlurIsPrevented = true;
 
+      const clearButton = this.$.input ? this.$.input.$.clearButton : this._clearElement;
```

Comment:
BTW the `_clearElement`-style accessors were introduced as an abstraction from the DOM implementation. The actual DOM location of the clear button depends on the component implementation and is different in regular and `-light` components. With this change, we handle that difference inside the shared mixin code, which is slightly more dirty.

Since the `vaadin-combo-box-light` does not need to access `this.$.input`, it would rather make sense to keep using `_clearElement` throughout the mixin code, and apply the change in the regular `vaadin-combo-box.html` instead.

Of couse, would be best to remove all `clearButton` references from the mixin altogether. But if that’s not possible, would perhaps make sense to keep using `_clearElement` as an abstracted reference.

Reply:
Yeah I agree some parts of the code are now a bit more dirty. But so seems to be much of the combo box code. 😄

I was thinking about setting `_clearElement` to `this.$.input` in `vaadin-combo-box.html` but that would have required some other changes. Probably something like introducing a new property e.g. `_customClearElement` which would be used to handle the logic needed for a custom clear button and using `_clearElement` only for the cases which should be handled the same way for both the custom clear button and the clear button coming from `vaadin-text-field`.

Reply:
I was able to refactor this so that the mixin only references `_clearElement` instead of 
 `this.$.input`.